src/charm.py

```diff
@@ -119,7 +132,28 @@ def __init__(self, *args):
         self.client_relation = PgBouncerProvider(self)
         self.legacy_db_relation = DbProvides(self, admin=False)
         self.legacy_db_admin_relation = DbProvides(self, admin=True)
-        self.tls = PostgreSQLTLS(self, PEER_RELATION_NAME, [self.unit_pod_hostname])
+
+        self.k8s_service_name = f"{self.app.name}-service"
+        unit_name = self.unit.name.replace("/", "-")
+
+        self.tls = PostgreSQLTLS(
+            self,
+            PEER_RELATION_NAME,
+            [
+                self.unit_pod_hostname,
+                self.k8s_service_name,
+                f"{self.k8s_service_name}.{self.model.name}.svc.cluster.local",
+                unit_name,
+                f"{unit_name}.{self.app.name}-endpoints.{self.model.name}.svc.cluster.local",
+                self.app.name,
+                f"{self.app.name}.{self.app.name}-endpoints",
+                f"{self.app.name}.{self.app.name}-endpoints.{self.model.name}.svc.cluster.local",
+                f"{self.app.name}-endpoints",
+                f"{self.app.name}-endpoints.{self.model.name}.svc.cluster.local",
+                f"{self.app.name}.{self.model.name}.svc.cluster.local",
+                "127.0.0.1",
```

Comment:
I don't think the loopback needs to be part of the cert.

Reply:
the reason why i included loopback was to be able to connect to pgbouncer from within the container. however, will remove as this is not necessarily a production requirement

Reply:
removed loopback in https://github.com/canonical/pgbouncer-k8s-operator/commit/46b037d599cbddcda6fdaed26488c21bff6dc6d6